packages/cli/src/commands/connector/utils.ts

```diff
@@ -169,25 +174,69 @@ export const addConnectors = async (instancePath: string, packageNames: string[]
 
 const officialConnectorPrefix = '@logto/connector-';
 
+type PackageMeta = { name: string; scope: string; version: string };
+
 const fetchOfficialConnectorList = async () => {
-  const { stdout } = await execPromise(`npm search ${officialConnectorPrefix} --json`);
-  const packages = z
-    .object({ name: z.string() })
-    .transform(({ name }) => name)
-    .array()
-    .parse(JSON.parse(stdout));
+  // See https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#get-v1search
+  type FetchResult = {
+    objects: Array<{
+      package: PackageMeta;
+      flags?: { unstable?: boolean };
+    }>;
+    total: number;
+  };
 
-  return packages.filter((name) =>
-    ['mock', 'kit'].every(
-      (excluded) => !name.slice(officialConnectorPrefix.length).startsWith(excluded)
-    )
-  );
+  const fetchList = async (from = 0, size = 20) => {
+    const parameters = new URLSearchParams({
+      text: officialConnectorPrefix,
+      from: String(from),
+      size: String(size),
+    });
+
+    return got(
+      `https://registry.npmjs.org/-/v1/search?${parameters.toString()}`
+    ).json<FetchResult>();
+  };
+
+  const packages: PackageMeta[] = [];
+
+  // Disable lint rules for business need
+  // eslint-disable-next-line @silverhand/fp/no-let, @silverhand/fp/no-mutation
+  for (let page = 0; ; ++page) {
```

Comment:
Can we fetch npm registry results with a single request instead of using loops?

Reply:
The max size of one single request is 100, which could be a reasonable size that the number of total results could reach.